packages/app/src/cli/models/extensions/specification.ts

```diff
@@ -225,6 +236,57 @@ export class ExtensionInstance<TConfiguration extends BaseConfigType = BaseConfi
   get isJavaScript() {
     return Boolean(this.entrySourceFilePath?.endsWith('.js') || this.entrySourceFilePath?.endsWith('.ts'))
   }
+
+  get functionExtension(): FunctionExtension | undefined {
+    if (!this.isFunctionExtension) return undefined
+    return this as unknown as FunctionExtension
+  }
+
+  async buildStep(options: ExtensionBuildOptions) {
+    if (this.isThemeExtension) {
+      return buildThemeExtension(this, options)
+    } else if (this.isFunctionExtension) {
+      return buildFunctionExtension(this.functionExtension!, options)
+    } else if (this.features.includes('esbuild')) {
+      return buildUIExtension(this, options)
+    }
+
+    // Workaround for tax_calculations because they remote spec NEEDS a valid js file to be included.
+    if (this.type === 'tax_calculation') {
+      await touchFile(this.outputBundlePath)
+      await writeFile(this.outputBundlePath, '(()=>{})();')
+    }
+  }
+
+  async bundleStep(options: ExtensionBuildOptions, identifiers: Identifiers, bundleDirectory: string) {
```

Comment:
What about naming this function `bundleCode` so that it's easier to understand in what it differs from `bundleConfig`? At the moment I'd expect the `bundleStep` to contain all the bundling instructions.

Reply:
Makes senses, in fact `buildStep` and `bundleStep` do mostly the same thing, the bundle step doesn't actually bundle, it just builds in a different folder that is later included in a `zip`. the only extension that is different here is `theme` that does have a bundle phase. 
So I wasn't sure how to name this

Reply:
what about `buildForBundle`?